### ISSUE TITLE
How to use multi-gpu for training on one node.

### ISSUE THREAD

**Issue by Danliran** [horovod/horovod#1914]
**Environment:**
1. Framework: (TensorFlow, Keras, PyTorch, MXNet) pytorch
2. Framework version: 1.2.0
3. Horovod version: 0.19.1
4. MPI version:4.0.3rc4
5. CUDA version:10.2
6. NCCL version:2.4.8
7. Python version:3.6
8. OS and version:ubuntu18.04
9. GCC version:9.3

**Checklist:**
1. Did you search issues to find if somebody asked this question before?no
2. If your question is about hang, did you read [this doc] (https://github.com/horovod/horovod/blob/master/docs/running.rst)? no hang
3. If your question is about docker, did you read [this doc](https://github.com/horovod/horovod/blob/master/docs/docker.rst)? no docker
4. Did you check if you question is answered in the [troubleshooting guide](https://github.com/horovod/horovod/blob/master/docs/troubleshooting.rst)? yes

**Your question:**

One node with 2 GP. when i set -np 2, but two processor run the same GPU, another one is idel.
How to use both gpu for training?

command:
(py1.2gpu) root@taishan:~/py/horovod-master/examples# horovodrun -np 2 -H localhost:2 python pytorch_synthetic_benchmark.py
[1,0]<stdout>:taishan:60096:60247 [0] NCCL INFO Bootstrap : Using [0]lo:127.0.0.1<0>
[1,0]<stdout>:taishan:60096:60247 [0] NCCL INFO NET/Plugin : No plugin found (libnccl-net.so).
[1,0]<stdout>:taishan:60096:60247 [0] NCCL INFO NET/IB : No device found.
[1,0]<stdout>:taishan:60096:60247 [0] NCCL INFO NET/Socket : Using [0]lo:127.0.0.1<0>
[1,0]<stdout>:NCCL version 2.4.8+cuda10.1
[1,0]<stdout>:taishan:60096:60247 [0] NCCL INFO Setting affinity for GPU 0 to 03ffff
[1,0]<stdout>:taishan:60096:60247 [0] NCCL INFO Using 256 threads, Min Comp Cap 7, Trees enabled up to size -2
[1,0]<stdout>:taishan:60096:60247 [0] NCCL INFO comm 0xfffeac02d2e0 rank 0 nranks 1 cudaDev 0 nvmlDev 0 - Init COMPLETE
[1,0]<stdout>:Model: resnet50
[1,0]<stdout>:Batch size: 32
[1,0]<stdout>:Number of GPUs: 1
[1,0]<stdout>:Running warmup...
[1,1]<stdout>:taishan:60097:60248 [0] NCCL INFO Bootstrap : Using [0]lo:127.0.0.1<0>
[1,1]<stdout>:taishan:60097:60248 [0] NCCL INFO NET/Plugin : No plugin found (libnccl-net.so).
[1,1]<stdout>:taishan:60097:60248 [0] NCCL INFO NET/IB : No device found.
[1,1]<stdout>:taishan:60097:60248 [0] NCCL INFO NET/Socket : Using [0]lo:127.0.0.1<0>
[1,1]<stdout>:NCCL version 2.4.8+cuda10.1
[1,1]<stdout>:taishan:60097:60248 [0] NCCL INFO Setting affinity for GPU 0 to 03ffff
[1,1]<stdout>:taishan:60097:60248 [0] NCCL INFO Using 256 threads, Min Comp Cap 7, Trees enabled up to size -2
[1,1]<stdout>:taishan:60097:60248 [0] NCCL INFO comm 0xfffed802d0e0 rank 0 nranks 1 cudaDev 0 nvmlDev 0 - Init COMPLETE
[1,1]<stdout>:Model: resnet50
[1,1]<stdout>:Batch size: 32
[1,1]<stdout>:Number of GPUs: 1
[1,1]<stdout>:Running warmup...
[1,0]<stdout>:Running benchmark...
[1,1]<stdout>:Running benchmark...
[1,0]<stdout>:Iter #0: 51.4 img/sec per GPU
[1,1]<stdout>:Iter #0: 51.0 img/sec per GPU

+-----------------------------------------------------------------------------+
| NVIDIA-SMI 445.50       Driver Version: 445.50       CUDA Version: 11.0     |
|-------------------------------+----------------------+----------------------+
| GPU  Name        Persistence-M| Bus-Id        Disp.A | Volatile Uncorr. ECC |
| Fan  Temp  Perf  Pwr:Usage/Cap|         Memory-Usage | GPU-Util  Compute M. |
|=========================+======================|
|   0  Tesla T4            Off  | 00000000:02:00.0 Off |                    0 |
| N/A   52C    P0    26W /  70W |   1927MiB / 15109MiB |     17%      Default |
+-------------------------------+----------------------+----------------------+
|   1  Tesla T4            Off  | 00000000:04:00.0 Off |                    0 |
| N/A   51C    P0    27W /  70W |      3MiB / 15109MiB |      0%      Default |
+-------------------------------+----------------------+----------------------+

+-----------------------------------------------------------------------------+
| Processes:                                                                  |
|  GPU                  PID   Type   Process name                  GPU Memory |
|                                                                  Usage      |
|============================================
|    0                63107      C   python                            973MiB |
|    0                63108      C   python                            953MiB |
+-----------------------------------------------------------------------------+

**Comment by Danliran** [comment]
resolve, close it.

**Comment by catwt** [comment]
@Danliran Hello, I have the same problem as you, How did you solve it?

**Comment by Danliran** [comment]
one node with two GPU, command like this: horovodrun -np 2 -H localhost:2 python pytorch_mnist.py @catwt

**Comment by catwt** [comment]
> one node with two GPU, command like this: horovodrun -np 2 -H localhost:2 python pytorch_mnist.py @catwt

Thanks, I do this, but the two processes only user the first GPU( GPU 0), hvd.rank is always 0, do you know why?

**Comment by tgaddair** [comment]
Hey @catwt, this is usually caused by having mismatched versions of MPI at the time you build Horovod vs the time you run Horovod.  See: https://stackoverflow.com/questions/4039608/why-do-all-my-open-mpi-processes-have-rank-0

**Comment by Danliran** [comment]
> > one node with two GPU, command like this: horovodrun -np 2 -H localhost:2 python pytorch_mnist.py @catwt
> 
> Thanks, I do this, but the two processes only user the first GPU( GPU 0), hvd.rank is always 0, do you know why?

please make sure you have installed the correct version of openmpi and horovod with pytorch.
